test/tomography/test_gateset_tomography.py

```diff
@@ -0,0 +1,124 @@
+
+    @staticmethod
+    def hs_distance(A, B):
+        return sum([np.abs(x) ** 2 for x in np.nditer(A-B)])
+
+    def compare_gates(self, expected_gates, result_gates, labels, delta=0.2):
+        for label in labels:
+            expected_gate = expected_gates[label]
+            result_gate = result_gates[label]
+            msg = "Failure on gate {}: Expected gate = \n{}\n" \
+                  "vs Actual gate = \n{}".format(label,
+                                                 expected_gate,
+                                                 result_gate)
+            distance = self.hs_distance(expected_gate, result_gate)
+            self.assertAlmostEqual(distance, 0, delta=delta, msg=msg)
+
+    def run_test_on_basis_and_noise(self,
+                                    gateset_basis=StandardGatesetBasis,
+                                    noise_model=None,
+                                    noise_ptm=None):
+        labels = gateset_basis.gate_labels
+        gates = gateset_basis.gate_matrices
+
+        # apply noise if given
+        for label in labels:
+            if label != "Id" and noise_ptm is not None:
+                gates[label] = noise_ptm @ gates[label]
+        Fs = [gateset_basis.spam_matrix(label)
+              for label in gateset_basis.spam_labels]
+
+        # prepare the fitter
+        fitter = self.collect_tomography_data(shots=10000,
+                                              noise_model=noise_model,
+                                              gateset_basis=gateset_basis)
+
+        # linear inversion test
+        expected_gates = self.expected_linear_inversion_gates(gates, Fs)
+        result_gates = fitter.linear_inversion()
+        self.compare_gates(expected_gates, result_gates, labels)
+
+        # fitter optimization test
+        expected_gates = gates
+        result_gates = fitter.fit()
+        self.compare_gates(expected_gates, result_gates, labels)
+
+    def test_noiseless_standard_basis(self):
+        self.run_test_on_basis_and_noise()
+
+    def test_amplitude_damping_standard_basis(self):
+        gamma = 0.05
+        A0 = [[1, 0], [0, np.sqrt(1 - gamma)]]
+        A1 = [[0, np.sqrt(gamma)], [0, 0]]
+        noise_ptm = np.array([[1, 0, 0, 0],
+                              [0, np.sqrt(1-gamma), 0, 0],
+                              [0, 0, np.sqrt(1-gamma), 0],
+                              [gamma, 0, 0, 1-gamma]])
+        noise = QuantumError([([{'name': 'kraus',
+                                 'qubits': [0],
+                                 'params': [A0, A1]}], 1)])
+        noise_model = NoiseModel()
+        noise_model.add_all_qubit_quantum_error(noise, ['u1', 'u2', 'u3'])
+        self.run_test_on_basis_and_noise(noise_model=noise_model,
+                                         noise_ptm=noise_ptm)
+
+    def test_depolarization_standard_basis(self):
+        p = 0.05
+        noise_ptm = np.array([[1, 0, 0, 0],
```

Comment:
Same here

Reply:
Same as above. Here obtaining the PTM from the noise seems even more troublesome since I don't have an explicit Kraus representation without writing it myself.